pkg/operator/controllers/gameserverbuild_controller.go

```diff
@@ -224,12 +261,15 @@ func (r *GameServerBuildReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 				errCh <- err
 				return
 			}
+			newgs.Status.PrevState = mpsv1alpha1.GameServerStateInitializing
```

Comment:
this value is not persisted anywhere. As soon as the GameServer is created with `r.Create(ctx, newgs)` all fields in the `Status` subresource would have the default values (empty string for PrevState). Not sure if this impacts anything in the logic.

Reply:
Ah, I see, I've moved the line before .Create(), and have updated the gameserverbuild_controller.go to patch the gs's old state if need be.

Reply:
hm, even if you add that before .Create, I *think* that the status subresource can only be altered after the object has been created. Did you test it to see the behavior?

Reply:
That is correct. I've removed as it's not needed anyways. In testing the states moved from:
Current: Initializing, Previous: ""
Current: Standby, Previous: Initializing
Current: Standby, Previous: Standby

So the logic works as intended

Reply:
There are two more cases, not sure if it’s gonna have any impact on your code but mentioning them here just in case:

- Current: Active, Previous: StandingBy (this happens on allocation)
- Current StandingBy, Previous: “” (when the change happens too fast and the NodeAgent doesn’t have time to send the Initializing, or fails to do so)